src/Form/Control/TreeItemSelector.php

```diff
@@ -112,6 +120,36 @@ public function setTreeItems(array $treeItems)
         return $this;
     }
 
+    /**
+     * @param mixed $parentId
+     */
+    protected function addNodes(Model $model, $parentId = null): array
+    {
+        $result = [];
+
+        $nodeModel = (clone $model)->addCondition($this->parentIdField, $parentId);
+
+        foreach ($nodeModel as $node) {
+            if ($node->get($this->parentIdField) === $parentId) {
```

Comment:
why this cond?

Reply:
Because we have to set a new parentId condition on an unconditioned model, otherwise will have contradictory conditions